test/jdk/javax/swing/JTable/JTableRightAlignmentTest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Color;
+import java.awt.ComponentOrientation;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.AbstractTableModel;
+import javax.swing.UIManager;
+
+import javax.imageio.ImageIO;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * @test
+ * @key headful
+ * @bug 5108458
+ * @summary Test to check Right alignment of JTable data
+ * (Fix affects all L&F, test verifies only Metal L&F)
+ * @run main JTableRightAlignmentTest
+ */
+
+public class JTableRightAlignmentTest {
+    static JFrame frame;
+    static CustomTable table;
+    static boolean passed = true;
+    static String failureString;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
```

Comment:
Should this test run on all L&F? Since part of the fix occurs in BasicTableUI.

Reply:
I couldn't find a way to generalize the automatic test for all L&F, though I tested all manually.

Reply:
Since this PR touches Synth, there should a regression testcase for Synth L&F too..

Reply:
Ok.

Reply:
But the current logic in test doesn't hold good for few Look and Feel including Nimbus. Should I make the test as manual mentioning it to be tested against all Look and Feel?

Reply:
You can have two tests: for Metal and for Synth (Nimbus).

A manual test is an option too, but if you can automate each separate L&F, you should definitely do it: an automatic test will be run regularly and often.

Reply:
I have modified the test and now it can be tested for all Look and Feel.